cargo-dist/src/lib.rs

```diff
@@ -149,9 +150,44 @@ fn run_build_step(
             dist_graph, committish, prefix, target,
         )?),
         BuildStep::Extra(target) => run_extra_artifacts_build(dist_graph, target),
+        BuildStep::Updater(updater) => fetch_updater(dist_graph, updater),
     }
 }
 
+/// Fetches an installer executable and installs it in the expected target path.
+pub fn fetch_updater(dist_graph: &DistGraph, updater: &UpdaterStep) -> Result<()> {
+    let tmpdir = TempDir::new().into_diagnostic()?;
+
+    // Update this to work from releases, and to fetch prebuilt binaries,
+    // once this has been released at least once.
+    let mut cmd = Cmd::new(
+        "cargo",
+        format!("Fetch installer for {}", updater.target_triple),
+    );
+
+    // Install to a temporary path before moving it to the destination
+    cmd.arg("install")
+        .arg("--root")
+        .arg(tmpdir.path())
+        .arg("--git")
+        .arg("https://github.com/axodotdev/axoupdater")
+        .arg("--all-features")
+        .arg("--bin")
+        .arg("axoupdater");
```

Comment:
hmm in an ideal world we can fetch the latest release's prebuilt binaries? (with fallback to this for unbuilt platforms)

Reply:
Yes! Right now it's not done just because those binaries don't exist yet. This is the fallback until that happens. Likewise, it'll be built from the crates.io release instead of git.

Reply:
ignore me if i'm way off here, but i'm curious why we'd prefer the crates.io release over the git one (i think we often can't be sure there is a crates.io release, whereas we always know there is a git one)

Reply:
This is of axoupdater, rather than the user's own software. Because we publish axoupdater ourselves, we can know that we'll always have a stable version on crates.io.